internal/page/page.go

```diff
@@ -0,0 +1,132 @@
+package page
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/RHEnVision/provisioning-backend/internal/math"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ptr"
+	"github.com/rs/zerolog"
+)
+
+type (
+	ctxKeyType  int
+	limitOffset int
+)
+
+const (
+	offsetCtxKey ctxKeyType = iota
+	limitCtxKey
+)
+
+const (
+	defaultValueLimit  = 100
+	defaultValueOffset = 0
+)
+
+type Links struct {
+	Previous string `json:"previous"`
+	Next     string `json:"next"`
+}
+type Metadata struct {
+	Total int `json:"total"`
+}
+
+type Info struct {
+	Metadata Metadata `json:"metadata"`
+	Links    Links    `json:"links"`
+}
```

Comment:
The `Info` type name is pretty generic, I would suggest to find a better name for this.

Reply:
That is what me and @ezr-ondrej decided on slack after seeing examples in other services and online examples.

Reply:
I saw that, my point is that if this was not done or if you included links within the metadata structure, then the root type could have been named just "metadata" with its all associated functions/methods.